frontend/components/NctrlValueWidgets.jsx

```diff
@@ -193,29 +205,41 @@ export function NctrlValueSlider({ path, options, min, max, rerender, rerenderDe
           valueLabelDisplay="on"
           min={min}
           max={max}
-          step={0.001}
+          step={integer ? 1 : 0.001}
         />
       </div>
     );
   }
 }
 
-export function NctrlValueButtons({ path, buttons, rerender, rerenderDep }) {
+export function NctrlValueButtons({ path, buttons, selectedIndex, rerender, rerenderDep }) {
   const classes = useStyles();
 
   const nctrlValue = NctrlValue.of(path);
   const value = usePromiseGenerator(() => nctrlValue.value(), [nctrlValue, rerenderDep], '');
+  // Set first button as active by default
+  const [active, setActive] = useState(Object.keys(buttons)[0]);
 
   // the buttons are a map of label -> (formerValue) => nextValue
   return (
     <ButtonGroup size="large" className={classes.input}>
-      {Object.keys(buttons).map(label => {
+      {Object.keys(buttons).map((label, index) => {
         const clickFn = eval(buttons[label]);
+        const activeClass = active === label ? classes.active : '';
```

Comment:
is this really the way to do this?

Reply:
Have followed some docs and tutorials on React which were manipulating the classes. Have to check if there is a better way.

Reply:
couldnt you just pass disabeled to the button? or is there something else going on here?

Reply:
It's not disabled, just set active there.

Reply:
Maybe toggle buttons are the better choice there -> https://mui.com/components/toggle-button/

Reply:
yup sounds like it